### PR TITLE
Enable tpuvm pod support

### PR DESCRIPTION
**Pull request by JackCaoG**
This is to enable TPUVM support for pod.

TODO
1. verify pod setup other than v3-32 will work
2. verify if 2vm use case still works

Old pr in https://github.com/pytorch/xla/pull/2780

### CODE REVIEW

File: torch_xla/distributed/xla_dist.py
Comment (jysohn23) [inline torch_xla/distributed/xla_dist.py:586]: I think we can get rid of this now?

File: torch_xla/distributed/cluster.py
Comment (jysohn23) [inline torch_xla/distributed/cluster.py:389]: Could we just make this method take in a list of `service_workers` and basically just translate  those into `client_workers`? I don't want to block PR much longer so I can approve if needed.

File: torch_xla/distributed/xla_dist.py
Comment (jysohn23) [inline torch_xla/distributed/xla_dist.py:36]: Also, looks like this is from JAX's TPUVM setup script. To dedupe we can use `ClusterResolver.get_instance_metadata('accelerator-type')`

File: torch_xla/distributed/cluster.py
Comment (JackCaoG) [reply torch_xla/distributed/cluster.py:389]: I merge it with the `get_service_workers`, it might be a bit cleaner that way

File: torch_xla/distributed/xla_dist.py
Comment (jysohn23) [inline torch_xla/distributed/xla_dist.py:259]: `get_service_workers` is renamed so this shouldn't work?

File: torch_xla/distributed/xla_dist.py
Comment (jysohn23) [inline torch_xla/distributed/xla_dist.py:318]: ditto.

File: torch_xla/distributed/cluster.py
Comment (jysohn23) [inline torch_xla/distributed/cluster.py:339]: We have tests for `cluster.py` in https://github.com/pytorch/xla/blob/master/test/test_xla_dist.py could you fix those as those are failing in CI?

**Review by jysohn23 — COMMENTED** [review]
Thanks Jack!

File: torch_xla/distributed/xla_dist.py
Comment (JackCaoG) [reply torch_xla/distributed/xla_dist.py:259]: Ah.. the function I renamed is `ClusterResolver. get_service_workers `. We also have a `Cluster.get_service_workers` so this function still works.

File: torch_xla/distributed/xla_dist.py
Comment (JackCaoG) [reply torch_xla/distributed/xla_dist.py:318]: ditto

File: torch_xla/distributed/cluster.py
Comment (JackCaoG) [reply torch_xla/distributed/cluster.py:339]: will update and verify

**Comment by JackCaoG** [comment]
ehh.. it seems like trying to rebase messed up the commit again.. sigh

**Comment by JackCaoG** [comment]
Verify the tpuvm with pod up tp v3-512. Last thing I want to do before merging this pr is to make sure this doesn't break the 2vm use case. Will do that tomorrow.

File: torch_xla/distributed/xla_dist.py
Comment (jysohn23) [inline torch_xla/distributed/xla_dist.py:272]: Is this needed?

File: torch_xla/distributed/xla_dist.py
Comment (jysohn23) [inline torch_xla/distributed/xla_dist.py]: Is this an unused import?

**Review by jysohn23 — APPROVED** [review]
Thanks Jack!

File: torch_xla/distributed/xla_dist.py
Comment (JackCaoG) [reply torch_xla/distributed/xla_dist.py:272]: yup, `ShouldStartLocalServerBeforeInit ` will check `TPUVM_MODE`. We only want to start the local tf server early(before XrtComputationClient creation) in the tpu_pod setup.

Comment (jysohn23) [reply torch_xla/distributed/xla_dist.py:272]: Ah yes I see it thanks.

**Comment by JackCaoG** [comment]
Got a green run with v3-1024, will merge this pr for now and start back porting. I will let the xlml test to test the 2vm for me...

**Pull request merged**
